File: JugnooNewDriver/jugnooNewDriver/src/main/java/product/clicklabs/jugnoo/driver/datastructure/CustomerRideData.java
package product.clicklabs.jugnoo.driver.datastructure;

import android.content.Context;

import product.clicklabs.jugnoo.driver.GpsDistanceCalculator;

/**
 * Created by shankar on 5/30/16.
 */
public class CustomerRideData {
	private double distance = 0;
	private double haversineDistance = 0;
	private long startRideTime = System.currentTimeMillis();
	private long waitTime = 0;
	public CustomerRideData(){}

	public double getDistance(Context context) {
		double spDistance = GpsDistanceCalculator.getTotalDistanceFromSP(context);
		if(spDistance > distance){
			distance = spDistance;
		}
		return distance;
	}

	public void setDistance(double distance) {
		this.distance = distance;
	}




	public double getHaversineDistance() {
		return haversineDistance;
	}

	public void setHaversineDistance(double haversineDistance) {
		this.haversineDistance = haversineDistance;
	}






	public long getStartRideTime() {
		return startRideTime;
	}

	public void setStartRideTime(long startRideTime) {
		this.startRideTime = startRideTime;
	}






	public long getWaitTime() {
		return waitTime;
	}

	public void setWaitTime(long waitTime) {
		this.waitTime = waitTime;
	}

}

File: JugnooNewDriver/jugnooNewDriver/src/main/java/product/clicklabs/jugnoo/driver/datastructure/CustomerRideData.kt
package product.clicklabs.jugnoo.driver.datastructure

import android.content.Context
import product.clicklabs.jugnoo.driver.GpsDistanceCalculator
import product.clicklabs.jugnoo.driver.MeteringService

/**
 * Created by shankar on 5/30/16.
 */
class CustomerRideData {
    fun getDistance(context: Context?): Double {
        return GpsDistanceCalculator.getTotalDistanceFromSP(context)
    }

    fun getWaitTime(context: Context?): Long {
        return GpsDistanceCalculator.getWaitTimeFromSP(context)
    }

    fun getHaversineDistance(context: Context?): Double {
        return GpsDistanceCalculator.getTotalHaversineDistanceFromSP(context)
    }

    fun updateDistance(context: Context?, distance:Double){
        MeteringService.gpsInstance(context).updateTotalDistance(context, distance)
    }

}